### DISCUSSION TITLE
What Are The Breaking Changes of 6.x?

### DISCUSSION THREAD

**Discussion by Colorado4Wheeler** [fuatakgun/eufy_security#664] [General]
I always like to keep my extensions up-to-date but have been skipping the 6.x upgrade because of the statement "This is a breaking change and will probably require you to rework your existing automations and dashboards.  If you are happy with your current setup, there is no urgency to migrate but your integration might become inoperational in the future."

What are the main breaking changes that I need to be concerned about before upgrading to this?  I looked around here but couldn't find anything outlining the major changes that would cause my automations and dashboards to break so I can determine how I want to approach this update.  I could only find the release note #636 but it doesn't tell me specifically what the change is without me stepping through the code changes.

I also saw a statement made on another thread from the developer saying (and I'm paraphrasing) "I don't want to update this version any longer" and inviting another developer to step in - is this referring to keeping the 2.x version or is this project being abandoned in the near future?

Thank you!

**Comment by fuatakgun** [comment]
I am the only developer for this repository.

v2.x get messy by the time with hacks and new features.
v6.x is re-write of the integration.

What might break;
- renamed entities
- renamed services

Risk of sticking to v.2x is that, if something gets broken for some reason in the future, I will fix them on v6.x, not on v2.x.

**Reply by Colorado4Wheeler** [reply]
Thanks @fuatakgun both for writing this integration and for the fast response!  I assumed you were referring to the 2.x version as for a sunset but wanted to confirm.  

So the breaking change is only if I have renamed entities from what the integration named them initially?  If so then I'm golden, I haven't messed with those names (I don't think...).

**Reply by fuatakgun** [reply]
If you are gonna move into v6.x, I suggest you to remove the old one and install the new one
there might be same entity names between old and new ones but their internal IDs are different, so they will be created as duplicate.

I think, on average, a user can handle the post-migration issues in 10 minutes

**Reply by Colorado4Wheeler** [reply]
I dove right in and, yes, had to clean up duplicates and then rename the duplicates to be the old names and adjust some things that were inputs and changed to sensors, it wasn't really as bad as I thought it would be but even with only a few automations the process took 30 minutes or so to fix.  Overall this update looks really nice, so many new sensors and configs.

Just for the next person that has similar questions (my only Eufy item is a dual doorbell):

- Duplicates a dozen or so entities with the "_2" entity name, an easy fix
- Renders about 24 entities from 2.x invalid and they need to be removed
- Changes the type on a few previous "pseudo-sensors" that were input domains and they became sensor domains - meaning you will have to remove any dashboard items and re-add them since you cannot change the domain
- Changes the state values on a lot of entities.  Where there were entities that may have been on/off, they have been changed to states like clear/detected and running/not running
- You get at least twice as many diagnostic entities, double the config entities and a handful of new sensor entities (including some I wasn't even aware the doorbell could do, such as loitering)

Solid update!

**Reply by fuatakgun** [reply]
Thanks for sharing your findings

**Reply by rucknapucknavitz** [reply]
Agree - it was an easy fix post-update. I didn’t initially follow advice and delete before updating, so had a lot of “_2” entities, which of course broke everything. I then deleted and reinstalled to get back to original entity naming conventions. The progress on this integration has been excellent - thank you!

**Comment by akuehner** [comment]
I didn't delete the old integration before updating, which did break everything. I removed the integration, then re-added it, which seemed to temporarily fix all the devices. But now they keep becoming unavailable and I can't understand why. They temporarily came back after a restart, but then went unavailable again. I'm using a different account than what I use to login to the Eufy app directly, so it shouldn't be an auth thing. Any ideas on next steps?

**Reply by akuehner** [reply]
Are there any potential fixes for this? My Eufy system is still unable to stay connected to HA since the major update.
